addon/components/paper-text-field.js

```diff
@@ -0,0 +1,76 @@
+import Component        from 'ember-component';
+import formFieldOptions from
+  'ember-railio-form-components/mixins/form-field-options';
+import get   from 'ember-metal/get';
+import set   from 'ember-metal/set';
+
+export default Component.extend(formFieldOptions, {
+
+  focusIn() {
+    if (typeof this.attrs.focusIn === 'function') {
+      this.attrs.focusIn();
+    }
+  },
+
+  focusOut() {
+    let _value = get(this, '_value');
+    if (typeof this.attrs.focusOut === 'function') {
+      this.attrs.focusOut();
+    }
+
+    if (typeof get(this, 'format') === 'function') {
+      _value = this.format(_value);
+      set(this, '_value', _value);
+    }
+
+    this.send('changed', _value);
+  },
+
+  keyUp(e) {
+    if (typeof this.attrs.keyUp === 'function') {
+      this.attrs.keyUp();
+    }
+    if (e.keyCode === 13) {
+      if (typeof this.attrs.enter === 'function') {
+        this.attrs.enter();
+      }
+    }
+
+    if (e.keyCode === 27) {
+      if (typeof this.attrs.escape === 'function') {
+        this.attrs.escape();
+      }
+    }
+  },
+
+  didReceiveAttrs() {
+    let value = this.getAttr('value');
```

Comment:
getAttr?

Reply:
this is an ember invention to also access attributes, I didn't change it